middlewarex/oidc.go

```diff
@@ -0,0 +1,100 @@
+package middlewarex
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+)
+
+type (
+	OIDCMiddlewareOption func(*oidcMiddleware)
+
+	oidcMiddleware struct {
+		authFailedHandler func(error) http.HandlerFunc
+		config            *oidc.Config
+		verifier          *oidc.IDTokenVerifier
+	}
+	oidcContextKey int
+)
+
+const idTokenContextKey oidcContextKey = iota
+
+func OIDC(ctx context.Context, issuer string, opts ...OIDCMiddlewareOption) func(next http.Handler) http.Handler {
+	provider, err := oidc.NewProvider(ctx, issuer)
+	if err != nil {
+		panic(err)
+	}
+
+	mw := oidcMiddleware{
+		authFailedHandler: oidcAuthFailed,
+	}
+
+	for _, opt := range opts {
+		opt(&mw)
+	}
+
+	if mw.config == nil {
+		mw.verifier = provider.Verifier(&oidc.Config{})
+	} else {
+		mw.verifier = provider.Verifier(mw.config)
+	}
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			authHeader := r.Header.Get("Authorization")
```

Comment:
Nit: I would pass that function result directly to `validateAuthHeader` instead of creating a dedicated variable.